Improve PDF text rendering & alignment

vdye opened the pull request:
This pull request fixes a whole bunch of things with respect to PDF formatting. Changes include:

- Adjusting alignment of the header so that the song name is left aligned, measure numbers are centered, and move numbers are right aligned.
- Adding a `Field` class to represent field dimensions (for now, we're not really using it; it'll become more relevant in a follow-up PR).
- Generally scaling and positioning text so that positions are relative to margins/text size/other elements, rather than absolute pixel values.
- Fixing text overflow so now we don't need to build two versions of Rank Panda (one for "instructions on page with drill image" and one for "instructions on separate page").
- Adding page numbers.

Note that, right now, there's no sideline margin around the drill image, so things look a bit cramped. This will be addressed in a follow-up PR (I've just been sitting on this one for months now and I feel like I need to merge it at this point).

Closes https://github.com/bigredbands/rank-panda/issues/25

VinShepp commented on src/main/java/org/bigredbands/mb/controllers/PDFGenerator.java:
Commit message missing half a sentence "rank names are bolded to distinguish from"

vdye replied:
Fixed! See https://github.com/bigredbands/rank-panda/pull/26/commits/87155a518a939e2fccb3f2a71143a035215fa8db

VinShepp commented on src/main/java/org/bigredbands/mb/controllers/PDFGenerator.java:
```diff
@@ -240,154 +210,92 @@ public void createPDF(DrillInfo drillInfo, File file) throws IOException {
+                            // the list of commands, NOT the values of the
+                            // CommandPairs. Two direct-to-point CommandPairs
+                            // with different destinations aren't considered
+                            // "equal", but will render identically in PDF
```
Does it make sense to have two ranks that have the same direct-to-point moves that start/end at different places receive the same instructions?  Or since they're going to different places should it remain different?

vdye replied:
Both of them will show up as "DTP `<count>`" in the instruction text, so I'd probably want to consolidate them there. Then again, we might want to consider adding destination information for direct to point moves?

I'll leave the TODO in the code for now, we can always revisit that question if/when someone adds a "fix".

VinShepp commented on src/main/java/org/bigredbands/mb/utils/PDFStringUtils.java:
```diff
@@ -11,4 +11,49 @@ public static float stringWidth(String str, PDFont font,
+            // Token doesn't fit on the line
+            if (line.isEmpty()) {
+                // Just put the token on its own line (even though it's too long)
+                strs.add(token.trim());
```
There's got to be way to split the token across two lines if it's too long to fit on one.

vdye replied:
There probably is, I suppose someone could write a token that's way too long. I've added an issue to address this (FWIW, I never ran into this case on showcomm as far as I remember): https://github.com/bigredbands/rank-panda/issues/28

VinShepp reviewed (commented):
Added comments in-line.  Pretty much just fix the commit message and ponder my questions.  Will approve once finished.

vdye commented:
> Added comments in-line. Pretty much just fix the commit message and ponder my questions. Will approve once finished.

@VinShepp Addressed all your comments, plus removed an unused variable (see [here](https://github.com/bigredbands/rank-panda/compare/9e0b78b0c45bb68350b8dbdb867ab6534ad38dde..7bf73cc2d323e8261a30d1bf6721de6b64101fcf)).

VinShepp reviewed (approved):
Approved!

arslanaka commented:
Is this Issue still unfixed? if not I'll give it a try and fix it. let me know.
